plugins/hypervisors/vmware/src/com/cloud/hypervisor/vmware/resource/VmwareResource.java

```diff
@@ -3348,9 +3404,35 @@ protected Answer execute(StopCommand cmd) {
         // i.e., i-x-y. This is the internal VM name.
         VmwareContext context = getServiceContext();
         VmwareHypervisorHost hyperHost = getHyperHost(context);
+
         try {
             VirtualMachineMO vmMo = hyperHost.findVmOnHyperHost(cmd.getVmName());
+
             if (vmMo != null) {
```

Comment:
What about extracting this code from 3411-3429 to a method? It might also be good to Document it, and write some unit test cases

Reply:
Sure, I would do that.